lib/sqlalchemy/orm/query.py

```diff
@@ -1493,13 +1497,13 @@ def with_entities(
     @overload
     def with_entities(
         self, __ent0: _TCCA[_T0], __ent1: _TCCA[_T1], /
-    ) -> RowReturningQuery[Tuple[_T0, _T1]]:
+    ) -> RowReturningQuery[_T0, _T1]:
```

Comment:
**Federico Caselli** (CaselIT) wrote:

are all these overloads still needed?

Can't we use a single one with `*ent: _TCCA[Unpack[_Ts]]` or something similar?

If so this would apply to all other such overload chant, like session.query and the select constructors

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4986

Reply:
Unfortunately, it's not yet possible. Please, see - https://github.com/python/typing/discussions/1001

Reply:
This is a shame :(